WordPress/gutenberg
# Icon: Remove knobs in stories

Part of #35665

## What?

Remove the Knobs add-on from `Icon` stories. I've also converted the file to TypeScript since the component is already typed.

## Why?

This is now a blocker for supporting npm 8 (#46443).

## How?

Just a quick, minimal refactor.

## Testing Instructions

1. `npm run storybook:dev`
2. Go to the `Components / Icon` component. (Careful, there is also a `Icons / Icon` component 🙃)
3. The stories and controls should be covering the cases covered by the previous knobs. See inline code comments for intentional alterations.
